share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -380,5 +380,58 @@ function(sysbuild_module_call)
       endforeach()
     endif()
   endforeach()
+endfunction()
+
+# Usage:
+#   sysbuild_cache_set(VAR <variable> [APPEND [LIST]] <value>)
+#
+# This function will set the specified value of the sysbuild cache variable in
+# the CMakeCache.txt file which can then be accessed by images.
+# `VAR` specifies the variable name to set/update.
+#
+# The result will be returned in `<variable>`.
+#
+# Example use:
+#   sysbuild_cache_set(VAR ATTRIBUTES APPEND LIST battery)
+#     Will add the item `battery` to the `ATTRIBUTES` variable as a new element
+#     in the list in the CMakeCache.
+#
+# <variable>: Name of variable in CMake cache.
+# APPEND:     If specified then will append the supplied data to the existing
+#             value (unless LIST is also specified, this will be treated as a
+#             string, no space character is added between appends).
+# LIST:       Will treat the append item as a list and add the new data as a
+#             new element to the list.
+# <value>:    Value to set/update.
+function(sysbuild_cache_set)
+  cmake_parse_arguments(VARS "APPEND;LIST" "VAR" "" ${ARGN})
+
+  if(NOT VARS_UNPARSED_ARGUMENTS AND VARS_APPEND)
+    message(FATAL_ERROR "sysbuild_set(VAR <var> APPEND ...) missing required arguments to append")
```

Comment:
The first if checks if the append option is set and disallows no argument but allows set with no argument so is equivalent to `set(ABC)`, the LIST APPEND disallows an empty parameter but is there really a need for an empty parameter in a list?

Reply:
>  but is there really a need for an empty parameter in a list?

you see all sorts of variants on how to set variables, and feed that into other function calls (both in / outside loops).
For example by conditionally set value on a var and then feed it to another call without a check, like here:
https://github.com/zephyrproject-rtos/zephyr/blob/82101f45e4996713c2b9f3b455a9ff593311ca5f/CMakeLists.txt#L950-L958
which is then used here regardless if defined or not:
https://github.com/zephyrproject-rtos/zephyr/blob/82101f45e4996713c2b9f3b455a9ff593311ca5f/CMakeLists.txt#L968
I know in this specific case there are also other values going into the final call.
It's more to show a variable use without knowing if it has actually been defined.

Or by having a return var populated by a function, for example:
```
special_function(IN ${input} OUT output)
sysbuild_cache_set(VAR CACHE_VAR APPEND LIST ${output})
```

in such a case it could be that `special_function()` returns nothing, some functions may not return anything under certain conditions.

If empty vars are not allowed users will have to do:
```
special_function(IN ${input} OUT output)
if(DEFINED output)
  sysbuild_cache_set(VAR CACHE_VAR APPEND LIST ${output})
endif()
```

Because of such use-cases, I see little reason to impose this restriction, especially because CMake's own `list(APPEND ...)` doesn't have such restriction, and I value consistency whenever possible.

Reply:
Have reworked it